### PR TITLE
Supplier admin temp

### PR DESCRIPTION
**Pull request by Deepakkumar-kaaylabs**

### CODE REVIEW

File: src/styles/supplier.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        .accordion-button {
+            background-color: #004b87;
+            color: #ffff;
+            height: '42px'
+        }
+
+        .accordion-button::after {
+            background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23ffff'%3e%3cpath fill-rule='evenodd' d='M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z'/%3e%3c/svg%3e");
+        }
```
Comment (MJyotheesh) [inline src/styles/supplier.scss:20]: images or svgs don't import in scss files

File: src/styles/supplier-details.scss [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+            .accordion-button {
+                background-color: #004b87;
+                color: #ffff;
+                height: '42px'
+            }
+
+            .accordion-button::after {
+                background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16' fill='%23ffff'%3e%3cpath fill-rule='evenodd' d='M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z'/%3e%3c/svg%3e");
+            }
```
Comment (MJyotheesh) [inline src/styles/supplier-details.scss:23]: images or svgs don't import in scss files

File: src/styles/lookups.scss [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+    height: 65px;
   }
+    .lookup-body-container{
+      overflow: scroll;
+      height: calc(96% - 65px);
+    }
   .title {
-    font-family: "Montserrat";
-    font-weight: "bold";
+    font-family: "Montserrat",sans-serif;
```
Comment (MJyotheesh) [inline src/styles/lookups.scss:15]: font family give it common

File: src/pages/supplier/supplier-profile.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+const SupplierProfile = (props: SupplierProfileProps) => {
+  const { supplierUid } = props;
+  const [supplierData, setSupplierData] = useState<any>(null);
+  const [loading, setLoading] = useState(false);
+  const fetchSupplierData = async () => {
+    try {
+      setLoading(true);
+      const data = await getSupplier({ supplierUid, userType: 'admin' });
```
Comment (MJyotheesh) [inline src/pages/supplier/supplier-profile.tsx:17]: get role from const file

File: src/pages/supplier/supplier-profile.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      setSupplierData(data.result[0]);
+      setLoading(false);
+    } catch (error) {
+      setLoading(false);
+    }
+  };
+  useEffect(() => {
+    fetchSupplierData();
+    return ()=>{}
```
Comment (MJyotheesh) [inline src/pages/supplier/supplier-profile.tsx:26]: no use of returning empty object

File: src/pages/supplier/supplier-datail-list.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      <Col className="col-12 supplier-details-card px-1">
+       <CardComponent dataObj={obj} schema={schema}/>
+      </Col>
+    );
+  };
+
+  return (
+    <div className="container-fluid supplier-details-list-container">
+      <div className="accordin-conponent h-100">
+        <Accordion activeKey={collapse ? "1" : ""}>
```
Comment (MJyotheesh) [inline src/pages/supplier/supplier-datail-list.tsx:66]: Accordian was repeating most of the places can we make it as generic

File: src/pages/supplier/index.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  const [supplierUid, setSupplierUid] = useState<string>();
+  const [supplierList, setSupplierList] = useState([]);
+  const [loading, setloading] = useState(true);
+  const fetchSupplierData = async () => {
+    try {
+      setloading(true);
+      const data = await getSupplier();
+      const result: any = data.result;
+      setSupplierList(data.result);
+      if (data.result.length) {
```
Comment (MJyotheesh) [inline src/pages/supplier/index.tsx:21]: grt

File: src/pages/supplier/index.tsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+  return supplierList.length ? (
+    <div className="container-fluid h-100 m-0 supplier-container px-1 px-xs-1">
+      <div className="row m-0 align-items-center supplier-header">
+        <div className="col supplier-title">Supplier</div>
+        <div className="col p-0 d-flex justify-content-end">
+          <input />
+        </div>
+      </div>
+      <div className="row m-0" style={{ height: "93%" }}>
```
Comment (MJyotheesh) [inline src/pages/supplier/index.tsx:50]: inline

File: src/pages/driver/index.tsx [hunk 9]
```diff
@@ -1,9 +1,90 @@
...
 const Driver = () => {
+const [driverData, setDriverData] = useState([]);
+const [loading, setLoading ] = useState(false);
+const [activeTab, setActiveTab] = useState(tabList[0]);
+
+  const fetchDriverData = async()=>{
+    try {
+      setLoading(true);
+      const res = await getDriversList({status:activeTab,userType:'admin'})
+      setDriverData(res.result);
```
Comment (MJyotheesh) [inline src/pages/driver/index.tsx:24]: get

File: src/pages/driver/index.tsx [hunk 10]
```diff
@@ -1,9 +1,90 @@
...
 
 const Driver = () => {
+const [driverData, setDriverData] = useState([]);
+const [loading, setLoading ] = useState(false);
+const [activeTab, setActiveTab] = useState(tabList[0]);
+
+  const fetchDriverData = async()=>{
+    try {
+      setLoading(true);
+      const res = await getDriversList({status:activeTab,userType:'admin'})
```
Comment (MJyotheesh) [inline src/pages/driver/index.tsx:23]: get from const file

File: src/pages/machinery/index.tsx [hunk 11]
```diff
@@ -1,9 +1,90 @@
...
+      <NoDataIcon />
+    </div>
+  );
+  const Spinner = () => (
+    <div className="d-flex h-100 w-100 justify-content-center align-items-center">
+      <div className="spinner-border" role="status">
+        <span className="visually-hidden">Loading...</span>
+      </div>
+    </div>
+  );
```
Comment (MJyotheesh) [inline src/pages/machinery/index.tsx:44]: spinner make it as generic

**Review by MJyotheesh — CHANGES_REQUESTED** [review]
Resolve comments

File: src/shared/constant-values.ts [hunk 12]
```diff
@@ -9,3 +9,11 @@ export const RegexpList = {
   phone: /^\d{10}$/,
   email: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
 };
+
+export const mobileScreenWidth = 500;
```
[the remark this answers is not in the document]
Comment (Deepakkumar-kaaylabs) [reply src/shared/constant-values.ts:13]: ok changed

File: src/pages/supplier/supplier-datail-list.tsx (hunk 6 above)
Comment (Deepakkumar-kaaylabs) [reply src/pages/supplier/supplier-datail-list.tsx:66]: will try

File: src/pages/supplier/supplier-details.tsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+  
+  return (
+    <div className="supplier-details-container px-0" ref={ref}>
+      <SupplierProfile supplierUid={supplierUid} />
+      <SupplierDetail
+        header="Vehicle List"
+        restApiDetails={{
+          api: getVehicleList,
+          params: { supplierUid, status: Status.APPROVED },
+        }}
```
[the remark this answers is not in the document]
Comment (Deepakkumar-kaaylabs) [reply src/pages/supplier/supplier-details.tsx:33]: I am using the same file for the driver and vehicle so I need to pass that from the parent only.

**Pull request closed without merging**